lib/pwned/password.rb

```diff
@@ -123,5 +124,31 @@ def hashed_password_prefix
     def hashed_password_suffix
       hashed_password[HASH_PREFIX_LENGTH..-1]
     end
+
+    # Make a HTTP GET request given the url and headers.
+    # Yields a `Net::HTTPResponse`.
+    def with_http_response(url, headers, &block)
+      uri = URI(url)
+
+      request = Net::HTTP::Get.new(uri)
+      request.initialize_http_header(headers)
+
+      Net::HTTP.start(uri.host, uri.port, use_ssl: true) do |http|
+        http.request(request, &block)
+      end
+    end
+
+    # Stream a Net::HTTPResponse by line, handling lines that cross chunks.
+    def stream_response_lines(response, &block)
+      last_line = ''
+
+      response.read_body do |chunk|
+        chunk_lines = (last_line + chunk).lines
+        # This could end with half a line, so save it for next time
+        last_line = chunk_lines.pop
+        chunk_lines.each(&block)
+      end
```

Comment:
Oh! I need to `yield last_line` after this block. I should write a spec for this 🤦‍♂